File: Level2/221110/JavaScript/RescueBoat.js
export default function solution(people, limit) {
  let answer = 0;
  people.sort((a, b) => b - a);

  for (let left = 0, right = people.length - 1; left <= right; left += 1) {
    answer += 1;

    if (limit - people[left] >= people[right]) {
      right -= 1;
    }
  }

  return answer;
}

File: Programmers/Level2/221109/Java/RescueBoat.java
import java.util.Arrays;

public class RescueBoat {
    public int solution(int[] people, int limit) {
        int rescueBoatNumber = people.length;

        int rescuedPerson = 0;

        int notRescuedPerson = people.length - 1;

        Arrays.sort(people);

        while(rescuedPerson < notRescuedPerson) {
            if(people[rescuedPerson] + people[notRescuedPerson] > limit) {
                notRescuedPerson -= 1;
                continue;
            }

            if(people[rescuedPerson] + people[notRescuedPerson] <= limit) {
                notRescuedPerson -= 1;
                rescuedPerson += 1;

                rescueBoatNumber -= 1;
            }
        }

        return rescueBoatNumber;
    }
}
